src/components/InputFormat.js

```diff
@@ -12,33 +13,33 @@ class InputFormat extends Component {
   static propTypes = {
     getRef: PropTypes.func,
     label: PropTypes.string.isRequired,
-    numbersOnly: PropTypes.bool,
+    maxLength: PropTypes.number,
     delimiter: PropTypes.string,
     occurance: PropTypes.number,
-    maxLength: PropTypes.number,
     required: PropTypes.bool,
     onChange: PropTypes.func
   };
 
   static defaultProps = {
     getRef: null,
-    numbersOnly: false,
+    maxLength: 9999,
     delimiter: "",
     occurance: 0,
-    maxLength: 9999,
     required: false,
     onChange: null
   };
 
+  state = {
+    requiredChar:
+      formatLength(this.props.maxLength, this.props.occurance, this.props.delimiter) || 0
```

Comment:
formatLength, needs to return an appropriate value to ensure the `|| 0` is used for the assignment